### PR TITLE
[CHARMAP] Improve keyboard usability

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-13806](https://jira.reactos.org/browse/CORE-13806)

## Proposed changes

- Use `IsDialogMessage` function in message loop to enable Tab.
- Add `WS_TABSTOP` style to `IDC_FONTCOMBO` control.
- Set focus on create.
- Handle `WM_KEYDOWN` and `WM_GETDLGCODE` messages.
- Add `UpdateCells`, `LimitCaretXY` `SetCaretXY`, `MoveUpDown`, and `MoveLeftRight` helper functions.
- Delete `DrawGrid`, and `DrawActiveCell` functions for simplicity.
- Add and delete some members in `MAP` structure.

## TODO

- [x] Do tests.

### CODE REVIEW

File: base/applications/charmap/charmap.c [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * FILE:        base/applications/charmap/charmap.c
  * PURPOSE:     main dialog implementation
  * COPYRIGHT:   Copyright 2007 Ged Murphy <gedmurphy@reactos.org>
- *
+ *              Copyright 2022 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
```
Comment (HBelusca) [inline base/applications/charmap/charmap.c:7]: I think this is more justified in map.c than here!

File: base/applications/charmap/charmap.c [hunk 2]
```diff
@@ -567,6 +567,8 @@ PanelOnCreate(HWND hWnd, WPARAM wParam, LPARAM lParam)
         }
     }
 
+    SetFocus(hCharmapDlg);
```
Comment (HBelusca) [inline base/applications/charmap/charmap.c:570]: Why?

File: base/applications/charmap/charmap.c [hunk 3]
```diff
@@ -710,6 +712,12 @@ wWinMain(HINSTANCE hInst,
                     break;
                 }
 
+                /* NOTE: CreateDialog needs IsDialogMessage call in message loop */
+                if (hCharmapDlg && IsDialogMessage(hCharmapDlg, &Msg))
+                    continue;
+                if (hAdvancedDlg && IsDialogMessage(hAdvancedDlg, &Msg))
+                    continue;
```
Comment (HBelusca) [inline base/applications/charmap/charmap.c:719]:
```suggestion
#ifndef REMOVE_ADVANCED
                if (hAdvancedDlg && IsDialogMessage(hAdvancedDlg, &Msg))
                    continue;
#endif
```

File: base/applications/charmap/map.c [hunk 4]
```diff
@@ -90,55 +89,64 @@ DrawGrid(PMAP infoPtr,
     }
 }
 
+static
+VOID
+UpdateCells(PMAP infoPtr)
+{
+    INT x, y, i = XCELLS * infoPtr->iYStart;
```
Comment (HBelusca) [inline base/applications/charmap/map.c:96]:
```suggestion
    INT x, y;
    INT i = XCELLS * infoPtr->iYStart;
```

File: base/applications/charmap/charmap.c (hunk 2 above)
Comment (katahiromz) [reply base/applications/charmap/charmap.c:570]: This panel of `PanelOnCreate` is not a dialog.
So, it doesn't auto-focus.
`hCharmapDlg` is the first control of the panel.

File: base/applications/charmap/charmap.c (hunk 1 above)
Comment (katahiromz) [reply base/applications/charmap/charmap.c]: Moved.

**Pull request merged**